Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy.swift

```diff
@@ -558,6 +611,8 @@ extension PathHierarchy {
                 )
                 return
             }
+
+            assert(child.parent == nil, "Nodes that already have a parent should not be added to a different parent.")
```

Comment:
I added this assertion to facilitate testing, but i left it in because tests seemed to pass with it present. I could take it out if you think it's a problem.